torch/csrc/jit/runtime/register_special_ops.cpp

```diff
@@ -292,19 +292,22 @@ RegisterOperators reg({
           aliasAnalysisFromSchema()),
 
     DEFINE_TORCH_TENSOR_OP(
-        float,
-        double,
-        at::native::scalar_tensor(
-            scalar_val,
-            typeMetaToScalarType(c10::get_default_dtype()),
-            c10::nullopt /* layout */,
-            at::kCPU,
-            c10::nullopt /* pin_memory*/))
-        DEFINE_TORCH_TENSOR_OP(int, int64_t, at::scalar_to_tensor(scalar_val))
+        bool,
```

Comment:
AFAICT this is (a) reordering bool to be first, and (b) adding int, int64_t option... I'm not too familiar with this code, can you explain why this is is needed? (Or did I miss something and there's additional changes here that I didn't notice?)

Reply:
here， I just reordered bool to be first. int，int64_t is the original code. (by the way, the formator is bad for these codes)
then `aten::tensor(true, 'cpu')` can generate bool type tensor after bool to int conversion was added.

> pytorch does not use mixed allow_conversions first round, allow_conversions=false, but `device='cpu'` can't match signature `Device? device=None`. second round, allow_conversion=True, bool type will convert to float/int, and `device='cpu'` can also match. but the signature sequence causes us to hit the first one which returns a float, it's unexpected.
> 
> ```
> aten::tensor.float(float t, *, ScalarType? dtype=None, Device? device=None, bool requires_grad=False) -> Tensor
> aten::tensor.int(int t, *, ScalarType? dtype=None, Device? device=None, bool requires_grad=False) -> Tensor
> aten::tensor.bool(bool t, *, ScalarType? dtype=None, Device? device=None, bool requires_grad=False) -> Tensor
> aten::tensor.complex(complex t, *, ScalarType? dtype=None, Device? device=None, bool requires_grad=False) -> Tensor
> aten::tensor(t[] data, *, ScalarType? dtype=None, Device? device=None, bool requires_grad=False) -> Tensor
> ```